### PR TITLE
for #344: add utm params to mozilla.org links

### PR DESCRIPTION
**Pull request by groovecoder**

### CODE REVIEW

File: views/layouts/default.hbs [hunk 1]
```diff
@@ -42,21 +42,21 @@
     <footer id="footer">
       <div class="section-wrapper">
         <div class="third">
-          <a class="footer-logo-wrapper" href="https://www.mozilla.org" target="_blank" rel="noopener"><img src="/img/moz-logo-bw-rgb.svg" alt="Mozilla Logo" /></a>
+          <a class="footer-logo-wrapper" href="https://www.mozilla.org?utm_source={{ SERVER_URL }}&utm_medium=referral&utm_campaign=fx_monitor_downloads&utm_content=footer-link" target="_blank" rel="noopener"><img src="/img/moz-logo-bw-rgb.svg" alt="Mozilla Logo" /></a>
```
Comment (pdehaan) [inline views/layouts/default.hbs]: Unrelated to this, but I just noticed seeing them side-by-side(tm), but why is it `{{ SERVER_URL }}` here, but `{{ serverUrl }}` above (see ./views/email/email_partials/email_footer.hbs:L17):

```html
<img src="{{ serverUrl }}/img/moz-bw-email.png" width="40" height="40" style="width: 40px; height: 40px;" />
```

File: views/email/email_partials/email_footer.hbs [hunk 2]
```diff
@@ -23,7 +23,7 @@
       <p style="padding: 0px 0px 3px 0px; margin: 0px; font-weight: 600; color: #000000;">Mozilla</p>
       <p style="padding: 0px 0px 3px 0px; margin: 0px; font-weight: 300; color: #5d5d5d;">331 E. Evelyn Avenue Mountain View CA 94041</p>
       <p style="padding: 0px; margin: 0px;">
-        <a href="https://www.mozilla.org/about/legal/" rel="noopener" style="text-decoration: none; color: #0060df;" >Legal</a> | <a href="https://www.mozilla.org/privacy/websites/" rel="noopener" style="text-decoration: none; color: #0060df;">Privacy</a>
+        <a href="https://www.mozilla.org/about/legal/?utm_source=email&utm_medium=email&utm_campaign=fx_monitor_downloads&content=email_footer" rel="noopener" style="text-decoration: none; color: #0060df;" >Legal</a> | <a href="https://www.mozilla.org/privacy/websites/?utm_source=email&utm_medium=email&utm_campaign=fx_monitor_downloads&content=email_footer" rel="noopener" style="text-decoration: none; color: #0060df;">Privacy</a>
```
Comment (pdehaan) [inline views/email/email_partials/email_footer.hbs]: Probably overthinking it, but would it make sense to put these into a HBS variable?

```hbs
<a href="https://www.mozilla.org/about/legal/?{{ utm_email_footer }}" rel="noopener" ...>
```

File: views/layouts/default.hbs [hunk 3]
```diff
@@ -42,21 +42,21 @@
...
         </div>
         <ul class="third">
           <li>Support</li>
           <li><a href="https://qsurvey.mozilla.com/s3/Firefox-Monitor-Feedback" target="_blank" rel="noopener">Give Feedback</a></li>
-          <li><a href="https://www.mozilla.org/about/legal/" target="_blank" rel="noopener">Terms</a></li>
-          <li><a href="https://www.mozilla.org/privacy/websites/" target="_blank" rel="noopener">Privacy</a></li>
+          <li><a href="https://www.mozilla.org/about/legal/?utm_source={{ SERVER_URL }}&utm_medium=referral&utm_campaign=fx_monitor_downloads&utm_content=footer-link" target="_blank" rel="noopener">Terms</a></li>
```
Comment (pdehaan) [inline views/layouts/default.hbs]: Do we want the full `{{ SERVER_URL }}` here, or should we strip the "https://" prefix? Does it need to be URL-encoded at all?

Same as above, should we convert it into some handy HBS variable that we can recycle a bit easier:

```hbs
<a href="https://www.mozilla.org/about/legal/?{{ utm_site_footer }}" target="_blank" rel="noopener">...</a>
```

File: views/layouts/default.hbs [hunk 4]
```diff
@@ -42,21 +42,21 @@
...
+          <li><a href="https://www.mozilla.org/privacy/websites/?utm_source={{ SERVER_URL }}&utm_medium=referral&utm_campaign=fx_monitor_downloads&utm_content=footer-link" target="_blank" rel="noopener">Privacy</a></li>
```
Comment (pdehaan) [inline views/layouts/default.hbs]: Looking at this (versus one of the email ones above) side-by-side, I'm a bit confused between `content=email_footer` versus `utm_content=footer-link`.

Both in the consistency of `email_footer` versus `footer-link` (underscores versus hyphens) as well as `content=...` vs `utm_content=...`:

```
https://www.mozilla.org/privacy/websites/?utm_source=email&utm_medium=email&utm_campaign=fx_monitor_downloads&content=email_footer
https://www.mozilla.org/privacy/websites/?utm_source={{ SERVER_URL }}&utm_medium=referral&utm_campaign=fx_monitor_downloads&utm_content=footer-link
```

| SITE | `utm_source` | `utm_medium` | `utm_campaign` | `utm_content`*
|------|--------------|--------------|----------------|----------------
| https://www.mozilla.org/privacy/websites/ | email | mail | fx_monitor_downloads | content=email_footer
| https://www.mozilla.org/privacy/websites/ | `{{ SERVER_URL }}` | referral | fx_monitor_downloads | utm_content=footer-link

File: views/email/email_partials/email_footer.hbs [hunk 5]
```diff
@@ -13,7 +13,7 @@
   </tr>
   <tr>
     <td valign="center" style="margin:0px; padding: 0px; height: 80px;">
-      <a rel="noopener" href="https://www.mozilla.org" style="margin: 0px; padding: 0px;">
+      <a rel="noopener" href="https://www.mozilla.org/?utm_source=email&utm_medium=email&utm_campaign=fx_monitor_downloads&content=email_footer" style="margin: 0px; padding: 0px;">
```
Comment (pdehaan) [inline views/email/email_partials/email_footer.hbs]: Per comment below, should these be `&content=...` or `&utm_content=...`?

File: views/partials/banners/download_firefox_banner.hbs [hunk 6]
```diff
@@ -7,7 +7,7 @@
       </div>
     </div>
     <div class="third">
-      <a href="https://www.mozilla.org/firefox/new/?utm_campaign=ffxmdownloads" class="button" target="_blank" rel="noopener">Download Firefox</a>
+      <a href="https://www.mozilla.org/firefox/new/?utm_source={{ SERVER_URL }}&utm_medium=referral&utm_campaign=fx_monitor_downloads&utm_content=download_banner" class="button" target="_blank" rel="noopener">Download Firefox</a>
```
Comment (pdehaan) [inline views/partials/banners/download_firefox_banner.hbs]: And just to round out my stupidity, do you want underscores (consistent w/ "email_footer" above) or hyphens (consistent w/ "footer-link" above):

`&utm_content=download_banner` versus "download-banner" 

per:

`&content=email_footer`
`&utm_content=footer-link`

File: views/layouts/default.hbs (hunk 1 above)
Comment (groovecoder) [reply views/layouts/default.hbs]: Good point - we should try to consolidate on the naming convention. I can update `controllers/user.js` to set the email context var to be `SERVER_URL` like we set the website context var.

File: views/email/email_partials/email_footer.hbs (hunk 2 above)
Comment (groovecoder) [reply views/email/email_partials/email_footer.hbs]: Good idea. HBS doesn't support variable assignments in templates themselves, and it feels like a mis-match to put this in the controllers.

I'm going to try to move these into partials.

File: views/layouts/default.hbs (hunk 3 above)
Comment (groovecoder) [reply views/layouts/default.hbs]: @peterGerman also suggests stripping to the hostname. I'm adding a new `UTM_SOURCE` app local that will take care of it.

**Comment by groovecoder** [comment]
Updated with lots of niceties.

File: views/email/email_partials/email_footer.hbs (hunk 2 above)
Comment (pdehaan) [reply views/email/email_partials/email_footer.hbs]: Sure, I could see a partial or in server.js along w/ other `app.locals.BLAH` scoped stuff:

```js
app.locals.UTM_SOURCE = url.parse(AppConstants.SERVER_URL).hostname;
app.locals.UTM_EMAIL_FOOTER = "?utm_source=email&utm_medium=email&utm_campaign=fx_monitor_downloads&utm_content=email-footer-link";
app.locals.UTM_SITE_FOOTER = `?utm_source=${app.locals.UTM_SOURCE}&utm_medium=referral&utm_campaign=fx_monitor_downloads&utm_content=site-footer-link`;
```

```hbs
<a class="footer-logo-wrapper" href="https://www.mozilla.org/{{ UTM_SITE_FOOTER }}" target="_blank" rel="noopener">...</a>

<a rel="noopener" href="https://www.mozilla.org/{{ UTM_EMAIL_FOOTER }}" style="margin: 0px; padding: 0px;">...</a>
```

PROS:
- 2 fewer partials
- keeps UTM_SOURCE and other UTM_* stuff a bit closer together

CONS:
- polluting app.locals.*

File: views/partials/banners/download_firefox_banner.hbs [hunk 7]
```diff
@@ -7,7 +7,7 @@
       </div>
     </div>
     <div class="third">
-      <a href="https://www.mozilla.org/firefox/new/?utm_campaign=ffxmdownloads" class="button" target="_blank" rel="noopener">Download Firefox</a>
+      <a href="https://www.mozilla.org/firefox/new/?utm_source={{ UTM_SOURCE }}&utm_medium=referral&utm_campaign=fx_monitor_downloads&utm_content=download_banner" class="button" target="_blank" rel="noopener">Download Firefox</a>
```
Comment (pdehaan) [inline views/partials/banners/download_firefox_banner.hbs]: I think the only difference betweenst this and the partial below is that final `&utm_content=` value ("download_banner" versus "site-footer-link"). But I'm not convinced it's worth adding variables to the template to get this slightly DRYer.

```hbs
{{>email_button href=SERVER_URL value="Scan Another Email"}}
```
&mdash; via https://github.com/mozilla/blurts-server/pull/351/files#diff-aab49f213881484ad2be8406eddb6081R37

**Comment by lesleyjanenorton** [comment]
@lesleyjanenorton is late to the party as usual and would like to thank the hosts as she does so love to press shiny green buttons. this looks great.

File: views/partials/banners/download_firefox_banner.hbs (hunk 7 above)
Comment (groovecoder) [reply views/partials/banners/download_firefox_banner.hbs]: Oh, good point. I'll file a follow-up TODO.

**Pull request merged**